pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -193,7 +193,12 @@ func (r *Reconciler) DeleteSecretFromProviders(ctx context.Context, ps *esapi.Pu
 		}
 		client, err := mgr.Get(ctx, storeRef, ps.Namespace, nil)
 		if err != nil {
-			return out, fmt.Errorf("could not get secrets client for store %v: %w", storeName, err)
+			// if store is not handled by this controller instance
+			if err.Error() == "can not reference unmanaged store" {
+				return out, nil
```

Comment:
If we have a PushSecret that references multiple Stores, We have a condition logic error here, as the code would stop before its actual desired state:

```
goodStore1
goodStore2
badStore     # code stops here
goodStore3 # this one never gets its content deleted.
```